### PR TITLE
Human-readable errors from C++ pyclustering

### PR DESCRIPTION
**Pull request by annoviko**

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -15,6 +15,15 @@ GENERAL CHANGES:
 
 CORRECTED MAJOR BUGS:
```
Comment (annoviko) [inline CHANGES:17]: - [ ] Add information about #635 and CLIQUE error handling.

File: ccore/include/pyclustering/interface/clique_interface.h [hunk 2]
```diff
@@ -45,18 +46,18 @@ enum clique_package_indexer {
...
- * @param[in] p_intervals: amount of intervals in each dimension.
- * @param[in] p_threshold: minimum number of objects that should be contained by non-noise block.
- *
- * @return  Returns pointer to cure data - clustering result that can be used for obtaining
- *           allocated clusters, representative points and means of each cluster.
- *
- */
+/*!
+
+@brief   Clustering algorithm CLIQUE returns allocated clusters.
```
Comment (annoviko) [inline ccore/include/pyclustering/interface/clique_interface.h:51]: - [ ] Update description to:

```
@brief Performs cluster analysis of an input data using CLIQUE algorithm.
```

File: ccore/include/pyclustering/interface/clique_interface.h [hunk 3]
```diff
@@ -45,18 +46,18 @@ enum clique_package_indexer {
...
+@details Caller should destroy returned clustering data using 'cure_data_destroy' when
+          it is not required anymore.
+
+@param[in] p_sample: input data for clustering.
+@param[in] p_intervals: amount of intervals in each dimension that defines amount of CLIQUE blocks.
+@param[in] p_threshold: minimum number of objects that should be contained by non-noise block.
```
Comment (annoviko) [inline ccore/include/pyclustering/interface/clique_interface.h:57]: - [ ] Update description to:

```
... by non-outlier CLIQUE block.
```

File: ccore/include/pyclustering/interface/clique_interface.h [hunk 4]
```diff
@@ -45,18 +46,18 @@ enum clique_package_indexer {
...
+
+@return  Returns pointer to cure data - clustering result that can be used for obtaining
```
Comment (annoviko) [inline ccore/include/pyclustering/interface/clique_interface.h:59]: - [ ] Correct description:

```
Returns pointer to CLIQUE data - clustering result that can be used to obtain allocated clusters, outliers, logical locations, points captured by each block, min and max spatial corners.
```

File: ccore/include/pyclustering/interface/pyclustering_package.hpp [hunk 5]
```diff
@@ -286,17 +294,28 @@ pyclustering_package * create_package(const std::size_t p_size, const TypeValue
 
 /*!
 
-@brief   Create pyclustering package using container that supports `std::begin`, `std::end` functions and incremental iterators.
+@brief   Create pyclustering package using pointer to one-dimensional container that supports `std::begin`, `std::end` functions and incremental iterators.
```
Comment (annoviko) [inline ccore/include/pyclustering/interface/pyclustering_package.hpp:297]: - [ ] Update description to:
```
... pointer to one-dimensional container with fundamental data...
```

File: ccore/include/pyclustering/interface/pyclustering_package.hpp [hunk 6]
```diff
@@ -357,4 +380,58 @@ pyclustering_package * create_package(const std::vector< std::vector<TypeObject>
...
 
    return package;
-}
\ No newline at end of file
+}
+
+
+/*!
+
+@brief   Create pyclustering package using raw string line value such as `char *`, `wchar_t *` including any cv-qualified variants.
```
Comment (annoviko) [inline ccore/include/pyclustering/interface/pyclustering_package.hpp:388]: - [ ] Remove `line value` from the description.

File: ccore/include/pyclustering/interface/pyclustering_package.hpp [hunk 7]
```diff
@@ -357,4 +380,58 @@ pyclustering_package * create_package(const std::vector< std::vector<TypeObject>
...
+@details All data from the raw string will be copied to the package.
+
+@param[in] p_message: string line that is used to create pyclustering container.
```
Comment (annoviko) [inline ccore/include/pyclustering/interface/pyclustering_package.hpp:391]: - [ ] Update description: `... to create pyclustering package.`

File: ccore/include/pyclustering/utils/traits.hpp [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+        typename Type::const_iterator,
+        decltype(std::declval<Type>().cbegin()),
+        decltype(std::declval<Type>().cend())
+    >
+> : std::is_fundamental<typename Type::value_type> { };
+
+
+/*!
+
+@brief   Removes pointer, `const`, `volatile` from type `Type` if there are have a place in the type.
```
Comment (annoviko) [inline ccore/include/pyclustering/utils/traits.hpp:117]: - [ ] Update description: `... if they have a place in the type`.

File: ccore/include/pyclustering/utils/traits.hpp [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+@tparam Type: a type to update.
+
+*/
+template <typename Type>
+using remove_cvp = std::remove_cv<typename std::remove_pointer<Type>::type>;
+
+
+/*!
+
+@brief   Helper type that removes pointer, `const`, `volatile` from type `Type` if there are have a place in the type.
```
Comment (annoviko) [inline ccore/include/pyclustering/utils/traits.hpp:128]: - [ ] Update description: `... if they have a place in the type.`.

File: ccore/tst/utest-pyclustering.cpp [hunk 10]
```diff
@@ -95,6 +98,50 @@ TEST(utest_pyclustering, package_pointer_list) {
...
+    std::wstring message = L"message wstring";
+    std::shared_ptr<pyclustering_package> package = std::shared_ptr<pyclustering_package>(create_package(message));
+
+    ASSERT_NE(nullptr, package);
+    ASSERT_EQ(message.size() + 1, package->size);
+    ASSERT_EQ(PYCLUSTERING_TYPE_WCHAR_T, package->type);
+    ASSERT_EQ(0, std::wcscmp(message.c_str(), (const wchar_t *)package->data));
+}
+
+
```
Comment (annoviko) [inline ccore/tst/utest-pyclustering.cpp:144]: - [ ] Add test with empty `std::string`, `std::wstring`, `const char *` and `const wchar_t *`.

File: pyclustering/cluster/clique.py [hunk 11]
```diff
@@ -26,6 +26,7 @@
 
 
 import itertools
+import warnings
```
Comment (annoviko) [inline pyclustering/cluster/clique.py:29]: - [ ] Return back to line 45, it is not used.

File: pyclustering/core/pyclustering_package.py [hunk 12]
```diff
@@ -112,12 +115,18 @@ class package_builder:
...
+        @details String data is packed as it is without encoding. If string encoding is required then it should be
+                  provided already encoded, for example in case of `utf-8`:
+        @code
+            encoded_string = "String to pack".encode('utf-8')
+            pyclustering_package = package_builder(encoded_string)
+        @endcode
         
         @param[in] dataset (list): Data that should be packed in 'pyclustering_package'.
-        @param[in] c_data_type (ctype.type): If specified than specified data type is used for data storing in package. 
+        @param[in] c_data_type (ctype.type): Data C-type that is used to store data in the package.
```
Comment (annoviko) [inline pyclustering/core/pyclustering_package.py:129]: - [ ] Update description to `...: C-type data that is...`

File: pyclustering/core/pyclustering_package.py [hunk 13]
```diff
@@ -232,6 +244,13 @@ def __create_package_numpy_matrix(self, dataset_package, dataset):
         return pointer(dataset_package)
 
 
+    def __create_package_string(self, dataset_package, string_utf8):
```
Comment (annoviko) [inline pyclustering/core/pyclustering_package.py:247]: - [ ] Cosmetic. String could have any encoding. Rename variable to `string_value`.
